overlord/ifacestate/handlers.go

```diff
@@ -691,15 +696,27 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 				// NOTE: we don't log anything here as this is a normal and common condition.
 				continue
 			}
-			ts, err := ConnectOnInstall(st, chg, task, plug.Snap.Name(), plug.Name, slot.Snap.Name(), slot.Name)
-			if err != nil {
-				task.Logf("cannot auto-connect slot %s to %s: %s", connRef.SlotRef, connRef.PlugRef, err)
-				continue
+
+			if err := checkConnectConflicts(st, chg, plug.Snap.Name(), slot.Snap.Name(), task); err != nil {
+				if _, retry := err.(*state.Retry); retry {
+					task.Logf("cannot auto-connect slot %s to %s: %s", connRef.SlotRef, connRef.PlugRef, err)
+					continue
+				}
+				return fmt.Errorf("auto-connect conflict check failed: %s", err)
 			}
-			autots.AddAll(ts)
+			newconns = append(newconns, connRef)
 		}
 	}
 
+	// Create connect tasks and interface hooks
+	for _, conn := range newconns {
+		ts, err := connect(st, task, conn.PlugRef.Snap, conn.PlugRef.Name, conn.SlotRef.Snap, conn.SlotRef.Name)
+		if err != nil {
+			return fmt.Errorf("internal error: auto-connect of %q failed: %s", conn, err)
```

Comment:
Can we drop `internal error:` ?

Reply:
At this point this is really something that shouldn't happen, and on such occasions we tend to use "internal error:" prefix (you'll find a lot of examples of that in the project).